ledger/participant-integration-api/src/main/scala/platform/store/cache/StateCache.scala

```diff
@@ -85,12 +89,13 @@ private[platform] case class StateCache[K, V](
     * @param key the key at which to update the cache
     * @param fetchAsync fetches asynchronously the value for key `key` at the current cache index
     */
-  def putAsync(key: K, fetchAsync: Long => Future[V])(implicit
+  def putAsync(key: K, fetchAsync: Offset => Future[V])(implicit
       loggingContext: LoggingContext
   ): Future[V] = Timed.value(
     registerUpdateTimer,
     pendingUpdates.synchronized {
-      val validAt = cacheIndex
+      // TODO LLP: Revisit sanity of this condition
+      val validAt = cacheIndex.getOrElse(Offset.fromHexString(Ref.HexString.assertFromString("FF")))
```

Comment:
Looks very wrong.

Reply:
That's why I added the TODO :) Cut some corners for the PoC where this edge case doesn't matter that much and it does the trick